lib/esyBuildEjectCommand.js

```diff
@@ -0,0 +1,234 @@
+      }) {
+        let {
+          target,
+          command,
+          dependencies
+        } = rule;
+        return {
+          type: 'rule',
+          name: `*** ${packageJson.name}: ${target} ***`,
+          target: `${packageJson.name}.${target}`,
+          dependencies,
+          exportEnv: ['SHELL', 'ESY__SANDBOX'],
+          command: command != null
+            ? outdent`
+              $(${normalizedName}__ENV)\\
+              cd $cur__root; \\
+              ${command}
+            `
+            : null
+        };
+      }
+
+      let isRootPackage = packageJson.name === packageDb.rootPackageName;
+      let allDependencies = collectTransitiveDependencies(packageDb, packageJson.name);
+
+      let buildEnvironment = PackageEnvironment.calculateEnvironment(
+        packageDb,
+        packageJson.name
+      );
+
+      // Produce macro with rendered findlib.conf content.
+      let findlibPath = allDependencies
+        .map(dep => installDir(dep, 'lib'))
+        .join(':');
+
+      prelude.push({
+        type: 'define',
+        name: `${normalizedName}__FINDLIB_CONF`,
+        value: outdent`
+          path = "${findlibPath}"
+          destdir = "${installDir(packageJson.name, 'lib')}"
+        `
+      });
+
+      // Produce macro with rendered package's environment.
+      prelude.push({
+        type: 'define',
+        name: `${normalizedName}__ENV`,
+        value: Makefile.renderEnv(buildEnvironment),
+      });
+
+      rules.push({
+        type: 'rule',
+        name: null,
+        target: buildDir(packageJson.name, 'findlib.conf'),
+        exportEnv: ['ESY__SANDBOX', `${normalizedName}__FINDLIB_CONF`],
+        command: outdent`
+          mkdir -p $(@D)
+          echo "$${normalizedName}__FINDLIB_CONF" > $(@);
+        `
+      });
+
+      rules.push(makePackageRule({
+        target: 'findlib.conf',
+        dependencies: [buildDir(packageJson.name, 'findlib.conf')],
+      }));
+
+      rules.push(makePackageRule({
+        target: 'shell',
+        command: outdent`
+          $SHELL \\
+            --noprofile \\
+            --rcfile <(echo 'export PS1="[$cur__name sandbox] $ "')
+          `,
+      }));
+
+      rules.push(makePackageRule({
+        target: 'clean',
+        command: 'rm -rf $cur__install $cur__target_dir'
+      }));
+
+      let dependencies = collectDependencies(packageDb, packageJson.name)
+                         .map(dep => `${dep}.build`);
+
+      if (packageJson.pjc && packageJson.pjc.build) {
+        let buildCommand = packageJson.pjc.build;
+        rules.push(makePackageRule({
+          target: 'build',
+          dependencies: [
+            `${packageJson.name}.findlib.conf`,
+            ...dependencies
+          ],
+          command: isRootPackage
+          ? outdent`
+            $(ESY__PREPARE_CURRENT_INSTALL_TREE)\\
+            ${buildCommand}
+          `
+          : outdent`
+            if [ ! -d "$cur__install" ]; then \\
```

Comment:
So this is super silly, I know. We need to proper build caching instead (using env and deps' env as key probably). But for now it is at least fast. Note that it still always rebuilds root package. Also there's `<pkg name>.rebuild` target in generated `Makefile` to force rebuild of the package.